demo/dashboard/client/src/pages/files/File.js

```diff
@@ -40,29 +43,36 @@ class File extends React.Component {
   }
 
   loadFile() {
-    this.setState({
-      fileLoading: true,
-    });
     const { match } = this.props;
     const computedKey = decodeURIComponent(match.params.id);
 
-    return window
-      .fetch(
-        `${demoServerEndpoint}/filemeta?key=${encodeURIComponent(computedKey)}`
-      )
-      .then(handleErrors)
-      .then(response => response.json())
-      .then(data =>
-        this.setState({
-          fileLoading: false,
-          file: data[0],
-        })
-      )
-      .catch(error => {
-        console.log(`An error happened: ${error.message}`);
-      });
+    this.setState(
+      {
+        fileLoading: true,
+      },
+      () => {
```

Comment:
Could you please take the callback into a separate named function => easier to read, easier to maintain and debug potentially.